.github/workflows/coverage-diff.yml

```diff
@@ -0,0 +1,130 @@
+              /<metrics/ {
+              covered_statements += $10;
+              total_statements += $8;
+            }
+            END {
+              if (total_statements > 0) {
+                coverage = (covered_statements / total_statements) * 100;
+                if (coverage > 100) {
+                coverage = 100;
+            }
+            print coverage;
+            } else {
+              print 0;
+            }
+          }' cov/xml/clover.xml)
+          echo "COVERAGE_PR=$COVERAGE_PR" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+
+      - uses: shivammathur/setup-php@c541c155eee45413f5b09a52248675b1a2575231
+        with:
+          php-version: 8.2
+          coverage: none
+          tools: composer:v2
+          extensions: xdebug
+      - name: Install Dependencies
+        run: composer install -q --profile --ignore-platform-reqs --no-interaction --no-ansi --no-scripts --no-suggest --prefer-dist
+      - uses: KengoTODA/actions-setup-docker-compose@4677f0d86d41e623c9c6e11e1d910976da297bc0
+        with:
+          version: '2.14.2'
+      - name: Set Permissions for Coverage Directory
+        run: |
+          mkdir -p cov/json cov/xml cov/html
+          chmod -R 777 cov
+      - name: "Create Empty env File for Docker"
+        run: touch .env
+      - name: PHPUnit for Base
+        run: docker-compose run phpunit --coverage-clover=cov/xml/clover.xml
+      - name: Calculate coverage for base branch
+        id: base_coverage
+        run: |
+          COVERAGE_BASE=$(if [ -f cov/xml/clover.xml ]; then
+          awk -F'"' '
+            /<metrics/ {
+            covered_statements += $10;
+            total_statements += $8;
+          }
+          END {
+            if (total_statements > 0) {
+              print (covered_statements / total_statements) * 100;
+            } else {
+              print 0;
+            }
+          }' cov/xml/clover.xml
+          else
+            echo 0
+          fi)
+          echo "COVERAGE_BASE=$COVERAGE_BASE" >> $GITHUB_ENV
+          
+      - name: Compare coverage
+        run: |
+          COVERAGE_DIFF=$(echo "$COVERAGE_PR - $COVERAGE_BASE" | awk '{printf "%.2f", $0}')
+          echo "Coverage PR: $COVERAGE_PR%"
+          echo "Coverage Base: $COVERAGE_BASE%"
+          echo "Coverage Diff: $COVERAGE_DIFF%"
+          
+          if (( $(echo "$COVERAGE_DIFF > 0" | awk '{print ($0 > 0)}') )); then
+            COVERAGE_MESSAGE=":green_circle: Coverage increased by $COVERAGE_DIFF%"
+          elif (( $(echo "$COVERAGE_DIFF < 0" | awk '{print ($0 < 0)}') )); then
+            COVERAGE_MESSAGE=":red_circle: Coverage decreased by ${COVERAGE_DIFF#-}%"
+          else
+            COVERAGE_MESSAGE=":yellow_circle: Coverage remained the same."
+          fi
+          
+          echo "COVERAGE_MESSAGE=$COVERAGE_MESSAGE" >> $GITHUB_ENV
+
+      - name: Post coverage comment to PR
```

Comment:
Please use the same action [as here](https://github.com/fingerprintjs/dx-team-toolkit/blob/main/.github/workflows/coverage-diff.yml#L75) to add the coverage report to the PR. IT sill reuse the same comment for each action call.

Reply:
This should be fixed by this commit
https://github.com/fingerprintjs/fingerprint-pro-server-api-php-sdk/pull/103/commits/3e2b8324be199dfd554c8a82d21fe2408219ab5b